src/Elliptic/Systems/BnsInitialData/BoundaryConditions/StarSurface.hpp

```diff
@@ -0,0 +1,105 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+#include <string>
+#include <vector>
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/Tags.hpp"
+#include "Domain/Tags/FaceNormal.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryConditionType.hpp"
+#include "Elliptic/Systems//BnsInitialData/Tags.hpp"
+#include "Options/Context.hpp"
+#include "Options/String.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+/// \endcond
+
+namespace BnsInitialData ::BoundaryConditions {
+
+/*!
+ * Impose StarSurface boundary conditions:
+ * \f[
+ *   n_i F^i = \frac{C}{\alpha^2} B^i n_i.
+ * \f]
+ * The boundary condition results from requiring the conservation
+ * equations be regular at the surface of the neutron star.  See \cite
+ * BaumgarteShapiro 15.79.
+ */
+class StarSurface : public elliptic::BoundaryConditions::BoundaryCondition<3> {
+ private:
+  using Base = elliptic::BoundaryConditions::BoundaryCondition<3>;
+
+ public:
+  static constexpr Options::String help =
+      "StarSurface boundary conditions  n_i F^i = C/square(alpha) B^i n_i .";
+
+  using options = tmpl::list<>;
+
+  StarSurface() = default;
+  StarSurface(const StarSurface&) = default;
+  StarSurface& operator=(const StarSurface&) = default;
+  StarSurface(StarSurface&&) = default;
+  StarSurface& operator=(StarSurface&&) = default;
+  ~StarSurface() override = default;
+
+  /// \cond
+  explicit StarSurface(CkMigrateMessage* m) : Base(m) {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(StarSurface);
+  /// \endcond
+
+  std::unique_ptr<domain::BoundaryConditions::BoundaryCondition> get_clone()
+      const override {
+    return std::make_unique<StarSurface>(*this);
+  }
+
+  explicit StarSurface(const Options::Context& context);
+
+  std::vector<elliptic::BoundaryConditionType> boundary_condition_types()
+      const override {
+    return {elliptic::BoundaryConditionType::Neumann};
+  }
+
+  using argument_tags =
+      tmpl::list<gr::Tags::Lapse<DataVector>,
+                 BnsInitialData::Tags::RotationalShift<DataVector>,
+                 BnsInitialData::Tags::EulerEnthalpyConstant,
```

Comment:
Add this tag also to `volume_tags` below so we don't have to store it on every element face.

Reply:
So I just add it to `volume_tags` and it will magically not be saved on every element face?

Reply:
It will not be _retrieved_ from the faces, but just directly from the DataBox. So you don't have to add `domain::Tags::Faces<EulerEnthalpyConstant>` to the DataBox, but just add `EulerEnthalpyConstant` directly.